src/passes/Memory64Lowering.cpp

```diff
@@ -98,6 +98,9 @@ struct Memory64Lowering : public WalkerPass<PostWalker<Memory64Lowering>> {
     }
     // This is visited last.
     memory->indexType = Type::i32;
+    if (memory->hasMax() && memory->max > Memory::kMaxSize32) {
+      memory->max = Memory::kMaxSize32;
+    }
   }
```

Comment:
This one seems like it could be tested in a new small module in `./test/passes/memory64-lowering_enable-memory64_enable-bulk-memory.wast` ?